tests/test_tags.py

```diff
@@ -0,0 +1,391 @@
+            headers={'X-Grouper-User': user.username})
+
+    tag = PublicKeyTag.get(session, name="tyler_was_here")
+
+    user = session.query(User).filter_by(username="testuser@a.co").scalar()
+
+    fe_url = url(base_url, '/permissions/grant_tag/{}'.format(tag.name))
+    resp = yield http_client.fetch(fe_url, method="POST",
+            body=urlencode({'permission': TAG_EDIT, "argument": "*"}),
+            headers={'X-Grouper-User': user.username})
+
+    assert resp.code == 200
+    tag = PublicKeyTag.get(session, name="tyler_was_here")
+    perm = Permission.get(session, TAG_EDIT)
+    assert len(tag.my_permissions()) == 1, "The tag should have exactly 1 permission"
+    assert tag.my_permissions()[0].name == perm.name, "The tag's permission should be the one we added"
+    assert tag.my_permissions()[0].argument == "*", "The tag's permission should be the one we added"
+
+    # Make sure trying to add a permission to a tag doesn't fail horribly if it's already there
+    user = session.query(User).filter_by(username="testuser@a.co").scalar()
+
+    fe_url = url(base_url, '/permissions/grant_tag/{}'.format(tag.name))
+    resp = yield http_client.fetch(fe_url, method="POST",
+            body=urlencode({'permission': TAG_EDIT, "argument": "*"}),
+            headers={'X-Grouper-User': user.username})
+
+    assert resp.code == 200
+
+
+
+@pytest.mark.gen_test
+def test_edit_tag(users, http_client, base_url, session):
+
+    user = session.query(User).filter_by(username="testuser@a.co").scalar()
+
+    perm = Permission(name=TAG_EDIT, description="Why is this not nullable?")
+    perm.add(session)
+    session.commit()
+
+    grant_permission(session.query(Group).filter_by(groupname="all-teams").scalar(), session.query(Permission).filter_by(name=TAG_EDIT).scalar(), "*")
+
+    fe_url = url(base_url, '/tags')
+    resp = yield http_client.fetch(fe_url, method="POST",
+            body=urlencode({'tagname': "tyler_was_here", "description": "Test Tag Please Ignore"}),
+            headers={'X-Grouper-User': user.username})
+
+    tag = PublicKeyTag.get(session, name="tyler_was_here")
+
+    assert tag.description == "Test Tag Please Ignore", "The description should match what we created it with"
+
+    user = session.query(User).filter_by(username="testuser@a.co").scalar()
+    fe_url = url(base_url, '/tags/{}/edit'.format(tag.id))
+    resp = yield http_client.fetch(fe_url, method="POST",
+            body=urlencode({"description": "Don't tag me bro"}),
+            headers={'X-Grouper-User': user.username})
+
+    assert resp.code == 200
+
+    tag = PublicKeyTag.get(session, name="tyler_was_here")
+
+    assert tag.description == "Don't tag me bro", "The description should have been updated"
+
+
+@pytest.mark.gen_test
+def test_permissions(users, http_client, base_url, session):
+
+    user = session.query(User).filter_by(username="testuser@a.co").scalar()
+
+    perm = Permission(name=TAG_EDIT, description="Why is this not nullable?")
```

Comment:
prob cause we want to know what a permission's intended use case is. the description can be an empty string fwiw.